### PR TITLE
Fixed some bugs and refactoring (ref. issue #59)

### PR DESCRIPTION
**Pull request by aleattene**
- Fixed some bugs (algorithm.py) 
- Updated function "battle" (views.py)
- Modified classes "Request" and "Response" (request.py, response.py)
- Modified class "Battle" (algorithm.py) 
(ref. issue #59)

### CODE REVIEW

File: battle/battle_package/algorithm.py [hunk 1]
```diff
@@ -2,104 +2,97 @@
...
-	def define_defender(report):
-		defender = {}
-		for key, value in report.items():
-			if key == "defender":
-				defender = value
-		return defender
+	def __init__(self, request):
+		self.attacker = request.challengers_armies['attacker']  # parameters of the attacker
+		self.defender = request.challengers_armies['defender']  # parameters of the defender
+		self.focus, self.advantage = Battle.deploy(self)
```
Comment (aldotele) [inline battle/battle_package/algorithm.py]: is `Battle.deploy(self)` the same as `self.deploy()`  ?

File: battle/battle_package/algorithm.py [hunk 2]
```diff
@@ -2,104 +2,97 @@
...
+		self.special_round = Battle.play_special_round(self)
```
Comment (aldotele) [inline battle/battle_package/algorithm.py]: same here

**Review by aldotele — APPROVED** [review]

File: battle/battle_package/algorithm.py (hunk 2 above)
Comment (aleattene) [reply battle/battle_package/algorithm.py]: `self.play_special_round()` is the correct syntax for instance method `play_special_round`

File: battle/battle_package/algorithm.py (hunk 1 above)
Comment (aleattene) [reply battle/battle_package/algorithm.py]: `self.deploy()` is the correct syntax for instance method `deploy`

**Pull request merged**